### PR TITLE
#10: Suspending copies of actions for Pagination

### PR DESCRIPTION
**Pull request by Tetraquark**
#10 - copies of Pagination functions with suspend modifier.

### CODE REVIEW

File: paging/src/commonMain/kotlin/dev/icerock/moko/paging/Pagination.kt [hunk 1]
```diff
@@ -40,22 +40,32 @@ class Pagination<Item>(
...
-            listMutex.unlock()
+            loadFirstPageSuspend()
+        }
+    }
+
+    suspend fun loadFirstPageSuspend() {
+        mEndOfList.postValue(false)
+        mNextPageLoading.postValue(false)
+
+        listMutex.lock()
```
Comment (Alex009) [inline paging/src/commonMain/kotlin/dev/icerock/moko/paging/Pagination.kt:52]: it should be before we change mEndOfList and mNextPageLoading

File: paging/src/commonMain/kotlin/dev/icerock/moko/paging/Pagination.kt [hunk 2]
```diff
@@ -65,76 +75,82 @@ class Pagination<Item>(
...
-                // flag
-                mNextPageLoading.value = false
-                // notify
-                nextPageListener(Result.success(newList))
-            } catch (error: Throwable) {
-                // flag
-                mNextPageLoading.value = false
-                // notify
-                nextPageListener(Result.failure(error))
+        listMutex.lock()
```
Comment (Alex009) [inline paging/src/commonMain/kotlin/dev/icerock/moko/paging/Pagination.kt:78]: it also should be before we change any values

File: paging/src/commonMain/kotlin/dev/icerock/moko/paging/Pagination.kt (hunk 1 above)
Comment (Tetraquark) [reply paging/src/commonMain/kotlin/dev/icerock/moko/paging/Pagination.kt]: Mutex lock was moved.

File: paging/src/commonMain/kotlin/dev/icerock/moko/paging/Pagination.kt (hunk 2 above)
Comment (Tetraquark) [reply paging/src/commonMain/kotlin/dev/icerock/moko/paging/Pagination.kt]: Mutex lock was moved.

**Pull request merged**